### PR TITLE
feat(image): add new features

### PR DESCRIPTION
**Pull request by buqiyuan**
First of all, thank you for your contribution! 😄

New feature please send pull request to feature branch, and rest to master branch. Pull request will be merged after one of collaborators approve. Please makes sure that these form are filled before submitting your pull request, thank you!

[[中文版模板 / Chinese template](https://github.com/vueComponent/ant-design-vue/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### This is a ...

- [x] New feature
- [x] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [x] Refactoring
- [ ] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### What's the background?

before：

https://user-images.githubusercontent.com/39730999/162632327-e8dc2689-418d-43be-8881-24446f6f2ab5.mp4

after：

https://user-images.githubusercontent.com/39730999/162632334-69b69855-4283-4a8c-b3c0-0f6f6a5e5881.mp4

### API Realization (Optional if not new feature)

> 1. Basic thought of solution and other optional proposal.
> 2. List final API realization and usage sample.
> 3. GIF or snapshot should be provided if includes UI/interactive modification.

### What's the effect? (Optional if not new feature)

> 1. Does this PR affect user? Which part will be affected?
> 2. What will say in changelog?
> 3. Does this PR contains potential break change or other risk?

### Changelog description (Optional if not new feature)

> 1. English description
> 2. Chinese description (optional)

### Self Check before Merge

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### Additional Plan? (Optional if not new feature)

> If this PR related with other PR or following info. You can type here.

### CODE REVIEW

**Review by tangjinzhou — CHANGES_REQUESTED** [review]

File: components/image/__tests__/__snapshots__/demo.test.js.snap [hunk 1]
```diff
@@ -3,8 +3,11 @@
 exports[`renders ./components/image/demo/basic.vue correctly 1`] = `
 <div class="ant-image" style="width: 200px;"><img class="ant-image-img" src="https://zos.alipayobjects.com/rmsportal/jkjgkEfvpUPVyRjUImniVslZfWPnJuuZ.png">
   <!---->
-  <!---->
+  <div class="ant-image-mask">
+    <div class="[object Object]-mask-info"><span role="img" aria-label="eye" class="anticon anticon-eye"><svg focusable="false" class="" data-icon="eye" width="1em" height="1em" fill="currentColor" aria-hidden="true" viewBox="64 64 896 896"><path d="M942.2 486.2C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 000 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM512 766c-161.3 0-279.4-81.8-362.7-254C232.6 339.8 350.7 258 512 258c161.3 0 279.4 81.8 362.7 254C791.5 684.2 673.4 766 512 766zm-4-430c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm0 288c-61.9 0-112-50.1-112-112s50.1-112 112-112 112 50.1 112 112-50.1 112-112 112z"></path></svg></span>Preview</div>
```
[the remark this answers is not in the document]
Comment (buqiyuan) [reply components/image/__tests__/__snapshots__/demo.test.js.snap]: 是的 这里修复了
image

File: components/image/demo/preview-group-visible.vue [hunk 2]
```diff
@@ -18,7 +18,7 @@ Preview a collection from one image.
 
 <template>
   <a-image
-    :preview="{ visible }"
+    :preview="{ visible: false }"
```
[the remark this answers is not in the document]
Comment (buqiyuan) [reply components/image/demo/preview-group-visible.vue:21]: `ant-design`那边是要禁用的
image

File: components/image/index.zh-CN.md [hunk 3]
```diff
@@ -32,6 +33,10 @@ cover: https://gw.alipayobjects.com/zos/antfincdn/D1dXz9PZqa/image.svg
   visible?: boolean;
   onVisibleChange?: (visible, prevVisible) => void;
   getContainer: string | HTMLElement | (() => HTMLElement);
+  src?: string;
+  mask?: VNode;
```
[the remark this answers is not in the document]
Comment (buqiyuan) [reply components/image/index.zh-CN.md:37]: 好的 当时这里我也犹豫了下

File: components/vc-image/src/Image.tsx [hunk 4]
```diff
@@ -215,12 +242,13 @@ const ImageInternal = defineComponent({
...
         },
       };
+
       return (
         <>
           <div
             class={wrappperClass}
             onClick={
-              preview && !isError.value
+              props.preview && !Object.is(previewVisible.value, false) && !isError.value
```
[the remark this answers is not in the document]
Comment (buqiyuan) [reply components/vc-image/src/Image.tsx:251]: 这样判断会好像会有点问题，`props.preview.visible`为`false`的`Image`应当被禁用预览

File: components/image/index.tsx [hunk 5]
```diff
@@ -14,11 +17,42 @@ const Image = defineComponent<ImageProps>({
...
+      const { preview } = props;
+      const imageLocale = configProvider.locale?.Image || defaultLocale.Image;
+
+      if (preview === false) {
+        return preview;
+      }
+      const _preview = typeof preview === 'object' ? preview : {};
+
+      return {
+        mask: slots.previewMask ? (
```
[the remark this answers is not in the document]
Comment (buqiyuan) [reply components/image/index.tsx:32]: 那我直接改成这样可以吗？
image

File: components/vc-image/src/Image.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (buqiyuan) [reply components/vc-image/src/Image.tsx:251]: `!Object.is(previewVisible.value, false)`应该还是要保留的，`props.preview.visible`为false的不应该调用`onPreview`，而是直接触发`click`

File: components/image/index.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (tangjinzhou) [reply components/image/index.tsx]: 可以

File: components/vc-image/src/PreviewGroup.tsx [hunk 6]
```diff
@@ -135,17 +135,20 @@ const Group = defineComponent({
       mousePosition.value = null;
     };
 
+    const onPreviewAfterClose = () => {
```
[the remark this answers is not in the document]
Comment (buqiyuan) [reply components/vc-image/src/PreviewGroup.tsx]: done

File: components/vc-image/src/Image.tsx [hunk 7]
```diff
@@ -95,9 +96,12 @@ const ImageInternal = defineComponent({
     const getPreviewContainer = computed(() => preview.value.getContainer);
 
     const isControlled = computed(() => previewVisible.value !== undefined);
-    const isShowPreview = ref(!!previewVisible.value);
+    const [isShowPreview, setShowPreview] = useMergedState(!!previewVisible.value, {
+      value: previewVisible,
+      onChange: onPreviewVisibleChange.value,
```
[the remark this answers is not in the document]
Comment (buqiyuan) [reply components/vc-image/src/Image.tsx]: 已更正

File: components/vc-image/src/Preview.tsx [hunk 8]
```diff
@@ -263,7 +294,7 @@ const Preview = defineComponent({
                 onClick={onClick}
                 key={type}
               >
-                <IconType class={iconClassName} />
+                {createVNode(IconType, { class: iconClassName })}
```
Comment (tangjinzhou) [inline components/vc-image/src/Preview.tsx:297]: cloneVNode

File: components/vc-image/src/PreviewGroup.tsx [hunk 9]
```diff
@@ -55,39 +117,70 @@ const Group = defineComponent({
...
+
+    watch(previewVisible, () => {
+      isShowPreview.value = !!previewVisible.value;
+    });
+    watch(isShowPreview, (val, preVal) => {
+      onPreviewVisibleChange.value(val, preVal);
+    });
+    watch(currentControlledKey, val => {
+      setCurrent(val);
+    });
```
Comment (tangjinzhou) [inline components/vc-image/src/PreviewGroup.tsx:146]: 应该需要 immediate: true   吧
你可以测试下，受控显示 current:  1

**Review by tangjinzhou — CHANGES_REQUESTED** [review]

File: components/vc-image/src/PreviewGroup.tsx (hunk 9 above)
Comment (buqiyuan) [reply components/vc-image/src/PreviewGroup.tsx]: 是的 这里疏漏了

File: components/vc-image/src/Image.tsx [hunk 10]
```diff
@@ -248,12 +281,13 @@ const ImageInternal = defineComponent({
               </div>
             )}
             {/* Preview Click Mask */}
-            {previewMask && canPreview.value && (
-              <div class={`${prefixCls}-mask`}>{previewMask}</div>
+            {slots.previewMask && canPreview.value && (
+              <div class={`${prefixCls}-mask`}>{slots.previewMask()}</div>
```
[the remark this answers is not in the document]
Comment (buqiyuan) [reply components/vc-image/src/Image.tsx:285]: 😅啊对 漏了

[the remark this answers is not in the document]
Comment (buqiyuan) [reply components/vc-image/src/Image.tsx]: 已补上

**Comment by tangjinzhou** [comment]
image
这两都没过

**Comment by buqiyuan** [comment]
> image 这两都没过

😂同步了下[rc-dialog](https://github.com/react-component/dialog/blob/master/src/Dialog/index.tsx) `rootClassName`，代码里面没提示缺少这个属性 没注意

**Comment by Fox-ABC** [comment]
这个版本什么时候会发布

**Pull request merged**